### PR TITLE
TRNs for different classes  + TRN debug command

### PR DESCRIPTION
**Pull request by qndel**
Created a way to add TRNs for classes + debug command to test TRNs
Made loading TRNs optional so we can just add TRN files later to make it work

### CODE REVIEW

**Comment by qndel** [comment]
@glebm Wondering if https://github.com/diasurgical/devilutionX/pull/4076/commits/6c6bb906f80874eaaaeabbe64209275a49482463 is fine, I'm not a template expert 😅 not sure if there won't be any conflict between 
```cpp
template <typename T, bool isOptional = false>
```
and
```cpp
template <typename T, std::size_t N>
```
since bool can be treated as size and the opposite.

I'm just hoping 
```cpp
template <typename T, std::size_t N>
void LoadFileInMem(const char *path, std::array<T, N> &data)
```
arguments for that function are distinguishable enough that there won't be any confusion
Do you have any idea how to handle it in a more elegant way? Ideally some type like bool that doesn't convert to a number :D

**Comment by glebm** [comment]
It's fine but perhaps let's add a `LoadFileInMem(SDL_RWops *, ...)` instead?

File: Source/player.cpp [hunk 1]
```diff
@@ -364,16 +366,20 @@ void StartWalk(int pnum, Displacement vel, Direction dir, bool pmWillBeCalled)
...
 	if (data == nullptr && gbQuietMode)
 		return;
 
-	const byte *directionFrames[8];
+	byte *directionFrames[8];
 	CelGetDirectionFrames(data.get(), directionFrames);
+	auto trn = GetClassTRN(player);
 	for (size_t i = 0; i < 8; i++) {
+		if (trn != nullptr) {
+			Cl2ApplyTrans(directionFrames[i], reinterpret_cast<std::array<uint8_t, 256> &>(trn[0]), (int)directionFrames[i][0]);
```
Comment (AJenbo) [inline Source/player.cpp:381]: Don't use c-style casting

File: Source/player.cpp [hunk 2]
```diff
@@ -364,16 +366,20 @@ void StartWalk(int pnum, Displacement vel, Direction dir, bool pmWillBeCalled)
...
 	if (data == nullptr && gbQuietMode)
 		return;
 
-	const byte *directionFrames[8];
+	byte *directionFrames[8];
 	CelGetDirectionFrames(data.get(), directionFrames);
+	auto trn = GetClassTRN(player);
 	for (size_t i = 0; i < 8; i++) {
+		if (trn != nullptr) {
+			Cl2ApplyTrans(directionFrames[i], reinterpret_cast<std::array<uint8_t, 256> &>(trn[0]), static_cast<int>(directionFrames[i][0]));
```
Comment (glebm) [inline Source/player.cpp:381]: Previously we didn't have a `Cl2ApplyTrans` here for warrior/rogue/etc.
Why apply these now? For modding?

**Review by glebm — APPROVED** [review]

File: Source/player.cpp (hunk 2 above)
Comment (glebm) [reply Source/player.cpp:381]: Nevermind, I've now read the PR description :smile:

File: Source/player.cpp [hunk 3]
```diff
@@ -364,16 +366,20 @@ void StartWalk(int pnum, Displacement vel, Direction dir, bool pmWillBeCalled)
...
 {
 	data = nullptr;
 	data = LoadFileInMem(path);
 	if (data == nullptr && gbQuietMode)
 		return;
 
-	const byte *directionFrames[8];
+	byte *directionFrames[8];
 	CelGetDirectionFrames(data.get(), directionFrames);
+	auto trn = GetClassTRN(player);
```
Comment (glebm) [inline Source/player.cpp:378]:
```suggestion
	std::unique_ptr<uint8_t[]> trn = GetClassTRN(player);
```

File: Source/player.cpp (hunk 2 above)
Comment (glebm) [inline Source/player.cpp:381]: This is a very unsafe cast!

Let's make `GetClassTRN` return an `std::array<uint8_t, 256>` instead.

**Review by glebm — CHANGES_REQUESTED** [review]

**Review by glebm — DISMISSED** [review]

**Review by glebm — APPROVED** [review]

**Pull request merged**